bootstrap.yml

```diff
@@ -16,8 +16,19 @@
       - name: load instackenv file
         include_tasks: tasks/load_instackenv.yml
 
+      - name: check if the infrared dir exists
+        stat:
+          path: "{{ infrared_dir }}"
+        register: infrared_directory
+
+      - name: check if the infrared workspace dir exist
+        stat:
+          path: "{{ infrared_workspaces_dir }}"
+        register: infrared_workspaces_directory
+
       - name: setup infrared
         include_tasks: tasks/setup_infrared.yml
+        when: infrared_directory.stat.exists == False and infrared_workspaces_directory.stat.exists == False
```

Comment:
the infrared setup task called only if both the dirs are absent. but if only one dir misses the script will fail while accessing the other dir.

Reply:
TASK [run tripleo-undercloud] ****************************************************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Unable to change directory before execution: [Errno 2] No such file or directory: '/home/masco/infrared'"}

this is the error I am getting while infrared dir is missing.

Reply:
@asyedham still this condition is unchanged. i will tell you how to reproduce the above-mentioned case. on the jumphost just delete the 'infrared_dir` manually and rerun the deployment. i think we need to change the condition to `OR`.

Reply:
@venkataanil I feel we need to call this with `OR` condition. what do you think?

Reply:
> @venkataanil I feel we need to call this with `OR` condition. what do you think?

Agree

Reply:
@masco @venkataanil I changed it OR condition can you please have look